phylanx/ir/node_data.hpp

```diff
@@ -205,59 +209,60 @@ namespace phylanx { namespace ir
 
 
         /// Access a specific element of the underlying N-dimensional array
-        T& operator[](std::ptrdiff_t index)
+        T& operator[](std::size_t index)
         {
-            return data_.data()[index];
+            std::size_t idx_m = index / data_.columns();
+            std::size_t idx_n = index % data_.rows();
+            return data_(idx_m, idx_n);
         }
         T& operator[](dimensions_type const& indicies)
         {
             return data_(indicies[0], indicies[1]);
         }
 
-        T const& operator[](std::ptrdiff_t index) const
+        T const& operator[](std::size_t index) const
         {
-            return data_.data()[index];
+            std::size_t idx_m = index / data_.columns();
+            std::size_t idx_n = index % data_.rows();
+            return data_(idx_m, idx_n);
         }
         T const& operator[](dimensions_type const& indicies) const
         {
             return data_(indicies[0], indicies[1]);
         }
 
-        using iterator = detail::node_value_iterator<T>;
-        using const_iterator = detail::node_value_iterator<T>;
-
         /// Get iterator referring to the beginning of the underlying data
-        iterator begin() const
+        blaze::DenseIterator<const T, true> begin() const
```

Comment:
Please define an embedded types `iterator` and `const_iterator` to expose the iterator-type used.